### DISCUSSION TITLE
Element files in root

### DISCUSSION THREAD

**Discussion by bsastregx** [lit/lit#3441] [Q&A] [answered]
Hi, I am new to Lit.

I have cloned the Typescript starter.
This starter comes with a dummy element (my-element.ts).
When I build with **npm run build** four files are generated on the project root folder:

- my-lit-project/my-element.d.ts
- my-lit-project/my-element.d.ts.map
- my-lit-project/my-element.js
- my-lit-project/my-element.js.map

I would expect this files to be in a separate folder, may be in a distribution folder, but not there, in the root.
Is this the expected behaivor?

![lit](https://user-images.githubusercontent.com/54677414/201150178-655630e9-9f8d-4828-bdbd-8f32620fd81b.PNG)

**Comment by justinfagnani** [comment, marked as answer]
Yeah, this is expected for that file layout. If you want your files in a folder after being built you can put them in a subfolder of `src/`.

The reason way do it this way here is to have simpler imports int he case where you have multiple elements in a project. So if you had `element-1` and `element-2`, users could import like:

```ts
import {ElementOne} from 'your-package/element-1.js';
```

instead of:

```ts
import {ElementOne} from 'your-package/dist/element-1.js';
```

**Reply by justinfagnani** [reply]
With package exports being support more and more, we'll be able to move these to a folder _and_ keep the simpler imports.

**Comment by bsastregx** [comment]
Hi @justinfagnani , 

Thanks for the quick answer.
I understand the simpler import, but be aware that is somewhat caothic to have all this files on the project root.
Imagine that I have ten components, I will have 40 files in the root project folder, this approach seems disorganized to me.
